File: Swift/contains-duplicate-ii.swift
class Solution {
    func containsNearbyDuplicate(_ nums: [Int], _ k: Int) -> Bool {
        var dict = [Int: Int]()

        for (index, value) in nums.enumerated() {
            if let idx = dict[value], index - idx <= k {
                return true
            }
            dict[value] = index
        }

        return false
    }
}

File: Python/contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        d = dict()

        for index, value in enumerate(nums):
            if value in d and index - d[value] <= k:
                return True
            d[value] = index

        return False
